### ISSUE TITLE
Overriding default previews layout

### ISSUE THREAD

**Issue by AlecRust** [github/view_component#178]
I have a hand-rolled styleguide of `actionview-component` components that I'm looking to migrate to  the built-in component previews introduced in #96.

I want to utilise [this gem's `ComponentsController`](https://github.com/github/actionview-component/blob/master/lib/railties/lib/rails/components_controller.rb) rather than maintain my own, but I'd like to override the layout for the index page at `localhost:3000/rails/components`. This is the layout in question:

https://github.com/github/actionview-component/blob/master/lib/railties/lib/rails/templates/rails/components/index.html.erb

Based on description in #96 I expected to be able to add this layout at `test/components/previews/layouts/application.html.erb` but this doesn't seem to get picked up.

Am I missing something? Where can I add this layout file, or do I need to set up a separate route if I want to achieve this?

**Comment by joelhawksley** [comment]
@juanmanuelramallo mind having a look at this?

**Comment by juanmanuelramallo** [comment]
Hey @AlecRust, the template is actually [being used manually in the controller](https://github.com/github/actionview-component/blob/master/lib/railties/lib/rails/components_controller.rb#L17). So if you want to override it I would suggest overriding the index controller action.

```ruby
# lib/overrides/overrides_components_controller.rb

class OverridesComponentsController < Rails::ComponentsController
  def index
    # These first two lines are just being copied over
    @previews = ActionView::Component::Preview.all
    @page_title = "Component Previews"
    render template: "layouts/application" # or any other template you want to use here
  end
end
```

And in order to use this controller you would need to auto load it in development:
```ruby
# config/application.rb
config.autoload_paths << Rails.root.join('lib/overrides')

# or if you want to use this on production
# config.eager_load_paths << Rails.root.join('lib/overrides')
```

And to make the router use this controller (as it's [done here](https://github.com/github/actionview-component/blob/master/lib/action_view/component/railtie.rb#L59))
```ruby
# config/routes.rb
get "/rails/components"       => "overrides_components#index", :internal => true
```

I haven't tested this. So let me know if this works please.

**Comment by AlecRust** [comment]
I'm afraid that doesn't work for me, I get an empty `<body>` at `/rails/components` URL. The logs say this file was rendered:

```
/Users/alec/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/actionview-component-1.7.0/lib/railties/lib/rails/templates/rails/components/index.html
```

I was hoping to avoid overriding your controller. Considering what's involved I should probably stick with my existing `StyleguideController` for now. I looked at using `ActionView::Component::Preview.all` for the components list, but when it came to the "show" part of different preview examples I'd have to duplicate a lot of #96.

Our own styleguide also keeps the examples alongside the other relevant component files and renders all the different types together at `/styleguide/component_name` instead of linking to `/styleguide/component_name/type_1`, `/styleguide/component_name/type_2` etc.

![screenshot](https://i.imgur.com/jtpzd1h.png)

**Comment by juanmanuelramallo** [comment]
I think this has been solved by #311. Users can now [set the default layout to use in previews](https://github.com/github/view_component#:~:text=You%20can%20also%20set%20a%20custom%20layout%20to%20be%20used%20by%20default%20for%20previews%20as%20well%20as%20the%20preview%20index%20pages%20via%20the%20default_preview_layout%20configuration%20option:)

**Comment by joelhawksley** [comment]
Marking as solved. Thanks (again!) @juanmanuelramallo
